kuma/wiki/tests/test_views_list.py

```diff
@@ -121,3 +145,201 @@ def test_revisions_request_invalid_pages(root_doc, client):
     limit_url = urlparams(url, limit='nonsense')
     resp = client.get(limit_url)
     assert resp.status_code == 200
+
+
+def test_list_no_redirects(redirect_doc, doc_hierarchy_with_zones, client):
+    url = reverse('wiki.all_documents', locale='en-US')
+    resp = client.get(url)
+    assert resp.status_code == 200
+    assert 'public' in resp['Cache-Control']
+    assert 's-maxage' in resp['Cache-Control']
+    assert 'text/html' in resp['Content-Type']
+    # There should be 4 documents in the English locale.
+    assert len(pq(resp.content).find('.document-list li')) == 4
+    assert redirect_doc.slug not in resp.content
+
+
+def test_tags(root_doc, client):
+    """Test list of all tags."""
+    root_doc.tags.set('foobar', 'blast')
+    url = reverse('wiki.list_tags', locale=root_doc.locale)
+    resp = client.get(url)
+    assert resp.status_code == 200
+    assert 'foobar' in resp.content
+    assert 'blast' in resp.content
+    assert 'wiki/list/tags.html' in [t.name for t in resp.templates]
+    assert 'public' in resp['Cache-Control']
+    assert 's-maxage' in resp['Cache-Control']
+
+
+@pytest.mark.tags
+@pytest.mark.parametrize('tag', ['foo', 'bar'])
+@pytest.mark.parametrize('tag_case', ['lower', 'upper'])
+@pytest.mark.parametrize('locale_case', ['root', 'trans'])
+def test_tag_list(root_doc, trans_doc, client, locale_case, tag_case, tag):
```

Comment:
I can't tell if this many parameters is clever or offensive. Maybe I'll have it figured out by 2019...